radix-engine/src/engine/actor.rs

```diff
@@ -1,51 +1,73 @@
 use crate::types::*;
 
+#[derive(Debug, Clone, Eq, PartialEq, TypeId, Encode, Decode)]
+pub enum FullyQualifiedMethodFn {
+    Scrypto {
+        package_address: PackageAddress,
+        blueprint_name: String,
+        ident: String,
+    },
+    Native(NativeMethodFnIdent),
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, TypeId, Encode, Decode)]
+pub struct FullyQualifiedMethod {
+    pub receiver: Receiver,
+    pub fn_ident: FullyQualifiedMethodFn,
```

Comment:
I think we only need the function ident here and the fully qualified path can be resolved from receiver and function identifier, no?

Reply:
Yup but the idea is that by the time the actor is created the fully qualified path has been resolved.

Reply:
That's fine. The reason I brought it up again is because of code like this:
``` 
(Receiver::Ref(RENodeId::Proof(proof_id)), NativeMethod::Proof(method)) => {
   Proof::main(proof_id, method, input, system_api).map_err(|e| e.into())
}
```
which can be simplified into 
```
Receiver::Ref(RENodeId::Proof(proof_id)) => {
   Proof::main(proof_id, method, input, system_api).map_err(|e| e.into())
}
```

The structure definition can prevent mismatching `Receiver` and `Method` statically.

And it's more object oriented which you've been advocating. :)